src/Package.vala

```diff
@@ -282,19 +282,19 @@ public class Eddy.Package : Object {
 
         int length = int.max (aparts.length, bparts.length);
         for (int i = 0; i < length; i++) {
-            int rc = strcmp (aparts[i], bparts[i]);
-            if (i == length - 1) {              
-                if (bparts[i] > aparts[i]) {
-                    return 1;
-                }      
-                else if (bparts[i] < aparts[i]) {
+            int a_num, b_num;
+            bool a_is_num = int.try_parse(aparts[i], out a_num);
+            bool b_is_num = int.try_parse(bparts[i], out b_num);
+
+            if (a_is_num && b_is_num) {
+                if (a_num < b_num) {
```

Comment:
if(a_num != b_num)
  return a_num - b_num;
  
  is also faster for cpu than comparisons